cmd/vat/analysis/target.go

```diff
@@ -0,0 +1,52 @@
+package analysis
+
+import (
+	"github.com/elrond-go/cmd/vat/utils"
+)
+
+// TargetStatus represents a target's state.
+type TargetStatus string
+
+// Enumerates the different possible state values.
+const (
+	New     TargetStatus = "NEW"
+	Scanned TargetStatus = "SCANNED"
+	Expired TargetStatus = "EXPIRED"
+)
+
+const (
+	NoCommand = iota
+	CHANGE_STATUS_TO_EXPIRED
+)
+
+// ActualStatus returns the status of a target.
+func (t DiscoveredTarget) ActualStatus() TargetStatus {
+	return TargetStatus(t.Status)
+}
+
+type DiscoveredTarget struct {
+	ID             uint
+	Protocol       string
+	Address        string
+	ConnectionPort string
+	Status         utils.TargetStatus
+}
+
+func MakeTarget(id uint, protocol string, address string, connectionPort string, status utils.TargetStatus) DiscoveredTarget {
+	return DiscoveredTarget{
+		ID:             id,
+		Protocol:       protocol,
+		Address:        address,
+		ConnectionPort: connectionPort,
+		Status:         status,
+	}
+}
+
+func containsTarget(haystack []DiscoveredTarget, needle DiscoveredTarget) bool {
+	for _, target := range haystack {
```

Comment:
Indeed using it as a map would be better. Can I keep it as slice for now, given the number of places where it is referenced and the impact it would have if I will change it?

Reply:
yes